src/test/java/learnositysdk/request/DataApiIT.java

```diff
@@ -0,0 +1,193 @@
+package learnositysdk.request;
+
+import java.util.UUID;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.json.JSONObject;
+import org.json.JSONArray;
+import org.apache.http.client.config.RequestConfig;
+
+import junit.framework.Test;
+import junit.framework.TestCase;
+import junit.framework.TestSuite;
+
+public class DataApiIT
+	extends TestCase {
+
+	static private String consumerKey = "yis0TYCu7U9V4o7M";
+	static private String consumerSecret = "74c5fd430cf1242a527f6223aebd42d30464be22";
+
+	private String baseUrl;
+	static Map<String,String> securityMap;
+	private RequestConfig requestConfig;
+	private Map request;
+	private DataApi dataApi;
+	static JSONObject response;
+	private JSONObject responseJson;
+
+	@Override
+	protected void setUp()
+		throws java.lang.Exception
+	{
+		String testEnv = System.getenv("ENV");
+		String testRegion = System.getenv("REGION");
+		String testVersion = System.getenv("VER");
+
+		baseUrl = buildBaseUrl(testEnv, testRegion, testVersion);
+
+		securityMap = new HashMap<String, String>();
+		securityMap.put("consumer_key", consumerKey);
+		securityMap.put("domain","localhost");
+
+		requestConfig = RequestConfig.custom().setSocketTimeout(40000).build();
+
+		request = new HashMap<String,String>();
+	}
+
+	public void testGetActivitites()
+			throws java.lang.Exception
+		{
+			String endpoint = baseUrl + "/itembank/activities";
+			System.out.println("Testing Data API call to " + endpoint + " with SET request");
+
+			JSONArray items = new JSONArray();
+			items.put("item_2");
+			items.put("item_3");
+
+			JSONObject data = new JSONObject();
+			data.put("items", items);
+
+			JSONObject activity = new JSONObject();
+			activity.put("status","published");
+			activity.put("description","My test description Title √");
+			activity.put("data", data);
+			activity.put("reference", UUID.randomUUID().toString());
+
+			JSONArray activities = new JSONArray();
+			activities.put(activity);
+
+			request.put("activities", activities);
+
+			DataApi dataApi = new DataApi(endpoint, securityMap, consumerSecret, request, "set");
+			response = dataApi.requestJSONObject();
+			responseJson = new JSONObject(response.getString("body"));
+
+			assertConsistentResponse(response, responseJson);
+		}
+
+	public void testGetItemsEmptyRemote()
+			throws java.lang.Exception
+		{
+			String endpoint = baseUrl + "/itembank/items";
+			System.out.println("Testing Data API call to " + endpoint + " with Remote object");
+
+			dataApi = new DataApi(endpoint, securityMap, consumerSecret);
+			response = dataApi.requestJSONObject();
```

Comment:
You could dry up the code a bit if you put the 5 lines above in their own function. They occur in almost all the tests (with additional request and and request type ('set' or 'get').

Reply:
Done, though it's not a big win.